routes/api/v1/users.js

```diff
@@ -6,35 +6,51 @@ var bcrypt = require('bcryptjs');
 router.post("/", function(req,res,next) {
   var email = req.body.email
   var password = req.body.password
-  if (req.body.passwordConfirmation === "") {
-    res.setHeader("Content-Type", "application/json");
-    res.status(409).send(`Must provide password confirmation.`)
-  }
-  else if (req.body.password === req.body.passwordConfirmation) {
-    bcrypt.genSalt(10, function(err,salt) {
-      bcrypt.hash(req.body.password, salt, function(err,hash) {
-        User.create({
-          email: email,
-          password: hash,
-          api_key: Math.random().toString(36).substring(2,8)
-          // the arg for toString is the radix which means what base (binary = 2); 36 includes letters
-          // substring takes the characters from index 2 to 8 not including 8
-        })
-        .then(user => {
-          res.setHeader("Content-Type", "application/json");
-          res.status(201).send(`${email} has been created`);
+  var confirm = req.body.passwordConfirmation
+
+  User.findOne({
+    where: {
+      email: email
+    }
+  }).then(result => {
+    if (result) {
+      res.setHeader("Content-Type", "application/json");
+      res.status(409).json({
+        error: `Email already exists in system.`
+      });
+    }
+    else {
+      if (confirm === "") {
+        res.setHeader("Content-Type", "application/json");
+        res.status(409).send(`Must provide password confirmation.`)
+      }
```

Comment:
Nice job including this additional validation!

Reply:
Thanks Corey!
I think I'll still had checks for whether or not the email is a valid format. I could also add a requirement for password length, but I haven't decided yet.